catalog/core/catalog-core-validationparser/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -296,6 +375,51 @@ private void parseValidators(Map<String, Object> root, Outer outer) {
                 .collect(toSet());
     }
 
+    @SuppressWarnings("unchecked")
+    private List<MetacardValidator> getMetacardValidators(
+            MetacardValidatorDefinition validatorDefinition) {
+        List<MetacardValidator> metacardValidators = new ArrayList<>();
+
+        MetacardValidator metacardValidator = null;
+        String validatorName = null;
+        Object metacardValidatorObj = validatorDefinition.arguments.get(VALIDATOR_PROPERTY);
+        if (metacardValidatorObj instanceof String) {
+            validatorName = (String) metacardValidatorObj;
+        }
+        if (validatorName != null) {
+            String metacardType = validatorDefinition.metacardType;
+            switch (validatorName) {
+            case REQUIRED_ATTRIBUTE_VALIDATOR_PROPERTY: {
+                Set<String> requiredAttributes = new HashSet<>();
+                Object requiredAttributesObj = validatorDefinition.arguments.get(
+                        REQUIRED_ATTRIBUTES_PROPERTY);
+                if (requiredAttributesObj instanceof List) {
+                    List<Object> requiredAttrObjList = (List) requiredAttributesObj;
+                    requiredAttributes = requiredAttrObjList.stream()
+                            .filter(String.class::isInstance)
+                            .map(String.class::cast)
+                            .collect(Collectors.toSet());
+                }
+
+                if (metacardType != null && CollectionUtils.isNotEmpty(requiredAttributes)) {
+                    metacardValidator = new RequiredAttributesMetacardValidator(metacardType,
+                            requiredAttributes);
+                } else {
```

Comment:
You could short-circuit at the top of this `case` block if the `metacardType` is null. Would save a little processing.